src/queries/swap.ts

```diff
@@ -0,0 +1,18 @@
+import { buildRequestBody, sendRequest } from './request';
+
+/**
+ * Makes an API request to the Swap API.
+ *
+ * @param method - The method to be called in the API.
+ * @param params - The parameters to be passed in the API request.
+ * @returns A promise that resolves to the result of the API request.
+ * @throws An error if the API request fails.
+ */
+export async function swapAPIRequest<T, V>(method: string, params: T[]): Promise<V> {
+  const body = buildRequestBody<T>(method, params);
+  const res = await sendRequest<T, V>(body);
+  if (res.error) {
+    throw new Error(res.error.message);
```

Comment:
Not a fan of all those throw errors, we need a better way to buble up and handling failure inside the library.

Reply:
I think there is a meta question to ask which is, what do we want to expose to the developer and what's the expeirence we want them to have.

Reply:
viem throws for a jsonrpc error:
https://viem.sh/docs/error-handling

Reply:
we can add a custom error type `SwapAPIError`

Reply:
this specific `throw` would be for a JSONRPC error (i.e. the request success and there is an `error` field on the response)